lib/api_auth/request_drivers/faraday.rb

```diff
@@ -56,7 +56,9 @@ def content_md5
       end
 
       def request_uri
-        uri = URI::HTTP.new(nil, nil, nil, nil, nil, @request.path, nil, @request.params.to_query, nil)
+        query_string = @request.params.map{|k,v| "#{k}=#{v}"}.join('&')
```

Comment:
I think using to_query was the right way to go to not run into ordering issues on Ruby 1.8.7
You can still do
```ruby 
query_string = @request.params.to_query
query_string = nil if query_string.empty?
```